src/data_structures/pauli_polynomial.rs

```diff
@@ -68,6 +68,31 @@ impl PauliPolynomial {
     pub fn angle(&self, i: usize) -> Angle {
         self.angles.read().unwrap()[i]
     }
+
+    pub fn visualize_pauli_polynomial(&self) {
+        let angles = self.angles.read().unwrap();
+        let string_angles = angles
+            .iter()
+            .map(|x| format!("{:.3}", x)) //force 3 decimal place for easy formatting
+            .collect::<Vec<String>>()
+            .join(" | ");
```

Comment:
`Itertools` has a `join` method for string iterators, that's probably more efficient than first allocating a vec and then creating a string.
```suggestion
            .join(" | ");
```

Reply:
fixed as you suggested